stdlib/public/Atomics/AtomicRepresentable.swift

```diff
@@ -0,0 +1,105 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public protocol AtomicRepresentable: AtomicProtocol, RawRepresentable
+where RawValue: AtomicProtocol, AtomicStorage == RawValue.AtomicStorage {}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension AtomicRepresentable {
+  //public typealias AtomicStorage = RawValue.AtomicStorage
+
+  @inlinable
+  public static func atomicStorage(for value: Self) -> RawValue.AtomicStorage {
+    return RawValue.atomicStorage(for: value.rawValue)
+  }
+
+  @inlinable
+  public static func deinitializeAtomicStorage(
```

Comment:
You could avoid this by requiring that AtomicStorage types be trivial.

Reply:
This is preparing for an atomic strong reference type, which will need a custom destructor even if the storage is trivial.

Reply:
But you're deinitializing the storage type, not the value type. That won't ever do anything.

Reply:
This is the teardown of the logical atomic value -- if `Self` is modeling a strong reference, then it needs to have a hook to release the value.

Reply:
Ah, okay. In that case, the implementation is wrong, since it's calling `deinitialize` on an `UnsafeMutablePointer<AtomicStorage>` and not `UnsafeMutablePointer<Self>` (or `UnsafeMutablePointer<RawValue>`).

Reply:
We don't have an atomic strong reference yet*. The existing atomic types are all trivial, so they could just as well have an empty implementation for `deinitializeAtomicStorage`.

* `UnsafeAtomicLazyReference` doesn't conform to `AtomicProtocol`, and (as of this moment) it has non-trivial storage.

Reply:
Ohhh, it's future-proofing. I get it now. Thanks for explaining it, sorry it took so long for me to understand!